sw/airborne/modules/display/max7456.c

```diff
@@ -113,8 +162,84 @@ uint8_t max7456_osd_status = OSD_UNINIT;
 uint8_t osd_enable = true;
 uint8_t osd_enable_val = OSD_IMAGE_ENABLE;
 uint8_t osd_stat_reg = 0;
-float home_dir = 0;
+uint32_t bat_capacity_left = 0;
+uint32_t max_flyable_distance_left = 0;
+
+float mag_course_deg = 0;
+float mag_heading_rad = 0;
+float gps_course_deg = 0;
+float home_dir_deg = 0;
+
+
+// Periodic function called with a frequency defined in the module .xml file
+void mag_compass(void)
```

Comment:
This function takes the raw magnetometer data and calculates the magnetic compass heading, is this already done?

Reply:
That shoud be directly `stateGetNedToBodyEulers_f()->psi` as I said. It is always true for rotorcraft, and for fixedwing it depends if you are using an AHRS with magnetometers or not (then, it is actually the ground course, I don't know if it is still applicable in your case)

Reply:
I tried with stateGetNedToBodyEulers_f()->psi and 
<define name="MODULE_HMC58XX_UPDATE_AHRS" value="TRUE"/>
in the airframe file but it does not work as i get a constant output of -1
What i did is to implement the stateGetNedToBodyEulers_f()->psi to rotorcrafts only.
If you want i can remove the function all together and use only the gps course.

Reply:
This is what i came up with 
  struct Int32Vect3 mag;
  struct Int32Vect3 mag_neutrals;
  VECT3_COPY(mag, imu.mag_unscaled);
  VECT3_COPY(mag_neutrals, imu.mag_neutral);
In order to buffer the imu structure as much as possible, the next step is to remove the mag_compass() function completely for fixed wings.
Let me know what you want me to do!

Reply:
I will accept the PR like this for now, you can't really expect a valid mag heading without mag, and if the DCM ahrs is fixed later to provide a reasonable output, this can be changed.